android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java

```diff
@@ -755,6 +765,12 @@ private DrmSessionManager initializePlayerDrm(ReactExoplayerView self) {
                 eventEmitter.error(getResources().getString(errorStringId), e, "3003");
                 return null;
             }
+        } else if (extension.equals(DOWNLOADED_CONTENT)) {
+            try {
+                drmSessionManager = buildDrmSessionManager(self.assetId);
+            } catch (Exception e) {
+                eventEmitter.error("Failed to setup downloads DRM", e);
```

Comment:
It seems like you would be better to throw this exception, rather than catch-and-continue, since without the DRM setup, it will just fail in the next step, probably with a much less clear error.

Reply:
Since `initializePlayerDrm` does not have any exception handling, we should handle exceptions here. It is straightforward to identify the issue as we have already sent an error message to the JS side using `eventEmitter.error("Failed to set up downloads DRM", e)`.  If needed, we can also include Logcat.

Reply:
if the problem is that throwing the error results in no logging, we should catch-log-and-throw.

It doesn't make sense to try and continue after this error, which is what is happening here - it will cause the player to continue to try and load without DRM, which will fail.

Reply:
this how the source code works: https://github.com/skytvnz/react-native-video/blob/881f5c66889e92aff0d07e7dba413b97d8053856/android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java#L748-L757

Reply:
The react-native-video project is badly coded, I think we know that by now. Particularly it's error handling (esp when it relates to DRM), but also memory leaks etc. 

If you throw the exception, it will arrive at this block, which seems fine.

```
 } catch (Exception ex) {
                self.playerNeedsSource = true;
                DebugLog.e("ExoPlayer Exception", "Failed to initialize Player!");
                DebugLog.e("ExoPlayer Exception", ex.toString());
                eventEmitter.error(ex.toString(), ex, "1001");
            }
```

Reply:
.. actually no, it gets lost, because it runs in this block, so the exception is lost:

```
es.execute(() -> {

}
```

My suggestion would be to wrap this entire execute block in a try-catch, because otherwise exceptions will be missed.

Reply:
Alternatively if you want to follow the existing (anti) pattern in this case - ok, since we can't really rewrite the whole library. Just keep in mind it's overall not the correct way to signal errors.